### PR TITLE
Add Blocking chart to gallery

### PR DESCRIPTION
**Pull request by RossKen**
### Type of PR

- [ ] BUG
- [ ] FEAT
- [ ] MAINT
- [X] DOC

### PR Checklist

- [X] Added documentation for changes
- [ ] Added feature to example notebooks or tutorial (if appropriate)
- [ ] Added tests (if appropriate)
- [ ] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

File: docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb [hunk 1]
```diff
@@ -140,15 +137,146 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### How to interpret the chart\n",
+    "\n",
+    "Blocking rules are order dependent, therefore each bar in this chart shows the **additional** comparisons generated ontop of the previous blocking rules.\n",
+    "\n",
+    "For example, the chart above shows and exact match on `surname` generates an additional 1351 comparisons. If we reverse the order of the `surname` and `first_name` blocking rules:"
```
Comment (ThomasHepworth) [inline docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb:172]:
```suggestion
    "For example, the chart above shows an exact match on `surname` generates an additional 1351 comparisons. If we reverse the order of the `surname` and `first_name` blocking rules:"
```

File: docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb [hunk 2]
```diff
@@ -157,7 +285,127 @@
...
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here, the total number of records pairs considered by the model have been reduced from 3,664 to 2,213.\n",
+    "\n",
+    "!!! note \"Further Reading\"\n",
+    "    :simple-readme: For a deeper dive on blocking, please refer to the [Blocking Topic Guides](../../topic_guides/blocking_rules.md).\n",
+    "\n",
+    "    :material-tools: For more on the blocking tools in Splink, please refer to the [Blocking API documentation](../../linkerbloc.md).\n"
```
Comment (ThomasHepworth) [inline docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb:408]: Neither link appears to work

File: docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb [hunk 3]
```diff
@@ -157,7 +285,127 @@
...
+    "            ),\n",
+    "        brl.exact_match_rule(\"surname\"),\n",
+    "        brl.exact_match_rule(\"email\")\n",
+    "    ]\n",
+    "}\n",
+    "\n",
+    "linker = DuckDBLinker(df, settings)\n",
+    "\n",
+    "linker.cumulative_num_comparisons_from_blocking_rules_chart()"
+   ]
```
Comment (ThomasHepworth) [inline docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb:396]: Do we want to show off the alternative syntax here?
```py
brs = [
        brl.and_(
            brl.exact_match_rule("first_name"),
            brl.exact_match_rule("dob"),
            ),
        brl.exact_match_rule("surname"),
        brl.exact_match_rule("email")
]
linker.cumulative_num_comparisons_from_blocking_rules_chart(brs)
```

Comment (ThomasHepworth) [reply docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb:396]: The `and_` composition can hopefully be removed before the end of the day too.

File: docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb (hunk 2 above)
Comment (RossKen) [reply docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb:408]: fixed

File: docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb (hunk 3 above)
Comment (RossKen) [reply docs/charts/cumulative_num_comparisons_from_blocking_rules_chart.ipynb:412]: Yea good point, I will add now

**Review by ThomasHepworth — APPROVED** [review]
Thanks Ross, this also looks good to go.

**Pull request merged**